dali/kernels/reduce/find_reduce_test.cu

```diff
@@ -83,29 +78,29 @@ class FindFirstLastTestGPU : public ::testing::Test {
 
     // Threshold = 3
     // Input 0:  0 1 2 1 0
-    //           ^^
+    //         ^^
     // Input 1:  0 1 2 3 4 4 3 2 1 0
     //                 ^     ^
     // Input 2:  0 1 2 3 4 3 2 1 0
     //                 ^   ^
     // Input 3:  0 1 2 3 2 1 0
     //                 ^^
-    // Input 4:  0 1 2 3 4 5 ... 1494 1495 1496 1497 1498 1499
-    //                 ^                   ^
-    ref_begin_.cpu()[0].data[0] = 0;
-    ref_length_.cpu()[0].data[0] = 0;
+    // Input 4:  0 1 2 3 4 5 ... 5 4 3 2 1 0
```

Comment:
Why changing the data here?

Reply:
The kernel now does something different, it finds first, and last separately. For computing length (or any other derived fields), we would need to launch a separate kernel. It was a tradeoff in order to make use of the reduction kernel infrastructure. We could probably make a kernel that still uses the reductions base and is able to do two reductions in one go, but I feel it will be quite convoluted.